.github/workflows/build-and-cache.yaml

```diff
@@ -24,17 +46,16 @@ jobs:
         uses: actions/checkout@v4
 
       - name: Install nix
-        uses: cachix/install-nix-action@v25
+        uses: cachix/install-nix-action@v27
         with:
-          install_url: https://releases.nixos.org/nix/nix-2.18.0/install
+          install_url: https://releases.nixos.org/nix/nix-2.22.1/install
```

Comment:
The versioning of the Nix package manager is confusing. On nixos.org the recommended version is 2.22.1, but in nixpkgs' nix the stable version is still 2.18.2

Reply:
Yeah it's quite unhelpful that. It is as you say, but we should be getting a new version of nixpkgs soon (a 2024 version) that will have the newer Nix in. 

I think as an end-user who is not running on NixOS, the latest Nix is fair game to use

Reply:
Yes, let's go with the one on the nix website.